contracts/strategies/strategy-axial-farm-base.sol

```diff
@@ -0,0 +1,145 @@
+            poolId,
+            address(this)
+        );
+        return amount;
+    }
+
+     receive() external payable {}
+
+    // Updated based on cryptofish's recommendation
+    function getHarvestable() external view returns (uint256, uint256) {
+        (uint256 pendingJoe, , , uint256 pendingBonusToken) = IMasterChefAxialV2(
+            masterChefAxialV3
+        ).pendingTokens(poolId, address(this));
+        return (pendingJoe, pendingBonusToken);
+    }
+
+    // **** Setters ****
+
+    function deposit() public override {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        if (_want > 0) {
+            IERC20(want).safeApprove(masterChefAxialV3, 0);
+            IERC20(want).safeApprove(masterChefAxialV3, _want);
+            IMasterChefAxialV2(masterChefAxialV3).deposit(poolId,_want);
+        }
+    }
+
+    function _withdrawSome(uint256 _amount)
+        internal
+        override
+        returns (uint256)
+    {
+        IMasterChefAxialV2(masterChefAxialV3).withdraw(poolId, _amount);
+        return _amount;
+    }
+
+    function _swapTraderJoe(
+        address _from,
+        address _to,
+        uint256 _amount
+    ) internal {
+        require(_to != address(0));
+
+        address[] memory path;
+
+        if (_from == wavax || _to == wavax) {
+            path = new address[](2);
+            path[0] = _from;
+            path[1] = _to;
+        } else {
+            path = new address[](3);
+            path[0] = _from;
+            path[1] = wavax;
+            path[2] = _to;
+        }
+
+        IJoeRouter(joeRouter).swapExactTokensForTokens(
+            _amount,
+            0,
+            path,
+            address(this),
+            now.add(60)
+        );
+    }
+
+    function _swapTraderJoeWithPath(address[] memory path, uint256 _amount)
+        internal
+    {
+        require(path[1] != address(0));
+
+        IJoeRouter(joeRouter).swapExactTokensForTokens(
+            _amount,
+            0,
+            path,
+            address(this),
+            now.add(60)
+        );
+    }
+
+    function _takeFeeAxialToSnob(uint256 _keep) internal {
+        address[] memory path = new address[](3);
+        path[0] = axial;
+        path[1] = wavax;
+        path[2] = snob;
+        IERC20(axial).safeApprove(joeRouter, 0);
+        IERC20(axial).safeApprove(joeRouter, _keep);
+        _swapTraderJoeWithPath(path, _keep);
+        uint256 _snob = IERC20(snob).balanceOf(address(this));
+        uint256 _share = _snob.mul(revenueShare).div(revenueShareMax);
+        IERC20(snob).safeTransfer(feeDistributor, _share);
+        IERC20(snob).safeTransfer(
+            IController(controller).treasury(),
+            _snob.sub(_share)
+        );
+    }
+
+     function _takeFeeWavaxToSnob(uint256 _keep) internal {
+        IERC20(wavax).safeApprove(pangolinRouter, 0);
+        IERC20(wavax).safeApprove(pangolinRouter, _keep);
```

Comment:
Just a small note: we're using pangolin's router instead of joe's, though it shouldn't pose an issue with the strat's function.